### PR TITLE
Added FanOut/FanIn pattern

### LINKED ISSUE #1627: Fan-Out/Fan-In pattern
https://dzone.com/articles/understanding-the-fan-out-fan-in-api-integration-p

### PR DESCRIPTION
**Pull request by karthikbhat13**
FanOut/FanIn Pattern

- The pattern refers to executing multiple instances of the activity function on multiple chunks in parallel
- Resolved #1627

Pull request description

- Added code that implements the design pattern and test cases

### CODE REVIEW

File: fanout-fanin/etc/fanout-fanin.urm.puml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    + fanOutFanIn(requests : List<SquareNumberRequest>, consumer : Consumer) : Long {static}
+  }
+  class SquareNumberRequest {
+    - LOGGER : Logger {static}
+    - number : Long
+    + SquareNumberRequest(number : Long)
+    + delayedSquaring(consumer : Consumer)
+  }
+}
+@enduml
```
Comment (ntantri) [inline fanout-fanin/etc/fanout-fanin.urm.puml:39]: Are there no relationships among "Associations" that can be depicted?

**Review by ntantri — CHANGES_REQUESTED** [review]

File: fanout-fanin/etc/fanout-fanin.urm.puml (hunk 1 above)
Comment (karthikbhat13) [reply fanout-fanin/etc/fanout-fanin.urm.puml:39]: I am kinda unaware of how it works. If one class is calling one more class as callback, is that an association?
I think it is right.

Comment (karthikbhat13) [reply fanout-fanin/etc/fanout-fanin.urm.puml:39]: I have updated!

Comment (ntantri) [reply fanout-fanin/etc/fanout-fanin.urm.puml:39]: Yes, the association bit is right and I think you have got It updated.

**Review by ntantri — APPROVED** [review]

**Pull request merged**